cf-serverd/server_classic.c

```diff
@@ -1142,36 +1142,28 @@ int BusyWithClassicConnection(EvalContext *ctx, ServerConnectionState *conn)
     switch (command)
     {
     case PROTOCOL_COMMAND_EXEC:
-        memset(args, 0, CF_BUFSIZE);
-        sscanf(recvbuffer, "EXEC %255[^\n]", args);
+    {
+        const size_t EXEC_len = strlen(PROTOCOL_CLASSIC[PROTOCOL_COMMAND_EXEC]);
+        /* Assert recvbuffer starts with EXEC. */
+        assert(strncmp(PROTOCOL_CLASSIC[PROTOCOL_COMMAND_EXEC],
+                       recvbuffer, EXEC_len) == 0);
 
-        if (!AllowedUser(conn->username))
-        {
-            Log(LOG_LEVEL_INFO, "REFUSAL due to non-allowed user");
-            RefuseAccess(conn, recvbuffer);
-            return false;
-        }
+        char *args = &recvbuffer[EXEC_len];
+        args += strspn(args, " \t");                       /* bypass spaces */
 
-        if (!AccessControl(ctx, CommandArg0(CFRUNCOMMAND), conn, false))
-        {
-            Log(LOG_LEVEL_INFO,
-                "REFUSAL due to denied access to requested object");
-            RefuseAccess(conn, recvbuffer);
-            return false;
-        }
+        Log(LOG_LEVEL_VERBOSE, "%14s %7s %s",
+            "Received:", "EXEC", args);
 
-        if (!MatchClasses(ctx, conn))
-        {
-            Log(LOG_LEVEL_INFO,
-                "REFUSAL due to failed class/context match");
-            Terminate(conn->conn_info);
-            return false;
-        }
+        bool b = DoExec2(ctx, conn, args,
+                         sendbuffer, sizeof(sendbuffer));
```

Comment:
Indeed. I think I'll leave it like that to honour the legacy, and hopefully we won't ever see a "3" version... ;-)